src/decode.rs

```diff
@@ -293,10 +303,11 @@ impl<'de> serde::Deserializer<'de> for Deserializer {
     }
 
     #[inline]
-    fn deserialize_bytes<V>(self, visitor: V) -> Result<V::Value>
+    fn deserialize_bytes<V>(mut self, visitor: V) -> Result<V::Value>
         where V: de::Visitor<'de>
     {
-        self.deserialize_seq(visitor)
+        let bytes = self.next_bytes()?;
+        visitor.visit_bytes(&bytes)
```

Comment:
This should call `visit_byte_buf` because there is no benefit of keeping ownership of the Vec\<u8\> within this function.

Reply:
Thanks for the tip @dtolnay!